File: vent-soft/src/main/java/com/qualitysales/ventsoft/Controllers/DTO/UserDTO.java
package com.qualitysales.ventsoft.Controllers.DTO;

import com.qualitysales.ventsoft.model.Role;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class UserDTO {

    private Integer id;
    @NotBlank
    private String name;
    private String lastName;
    private String login;
    private String password;
    private String code;
    private String email;
    private Boolean status;
    private Set<Role> roles;
}

File: vent-soft-front/src/app/business/user/user.dto.ts
import { RolesDTO } from "./user-form/roles.dto";

export interface UserDTO {
  id: number;
  name: string;
  lastName: string;
  login: string;
  password: string,
  code: string;
  email: string;
  roles: RolesDTO;
  status: boolean;
}
